src/graph/planner/plan/PlanNode.h

```diff
@@ -221,7 +221,7 @@ class PlanNode {
     return numDeps() == 2U;
   }
 
-  void setOutputVar(const std::string& var);
+  void setOutputVar(const std::string& var, const std::vector<std::string>& colNames);
```

Comment:
better to give the colNames parameter a default value, such as empty vector, so you don't have to update the usage of this function all the codebase <: